packages/query/src/graphql/plugins/PgConnectionArgFirstLastBeforeAfter.ts

```diff
@@ -34,7 +39,10 @@ export default (builder) => {
 
       addArgDataGenerator(function connectionFirstLastBeforeAfter({after, before, first, last, offset}) {
         return {
-          pgQuery: (queryBuilder) => {
+          pgQuery: (queryBuilder: QueryBuilder) => {
+            if (!unsafe) {
+              queryBuilder.limit(MAX_RECORD_COUNT);
```

Comment:
be careful don't conflict with the rest of codes, where `queryBuilder.first(first);` is set.